### PR TITLE
fix(gatsby): Vendor express-static so we can avoid loading html files from public during development

### PR DESCRIPTION
**Pull request by KyleAMathews**
Fixes https://github.com/gatsbyjs/gatsby/pull/12243#pullrequestreview-210895624

Works locally for reactjs.org which is a site that uses paths like
`/something/about-us.html`

screenshot 2019-03-05 14 18 11

### CODE REVIEW

File: packages/gatsby/src/commands/develop-static.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      res.end()
+      return null
+    }
+
+    var forwardError = !fallthrough
+    var originalUrl = parseUrl.original(req)
+    var path = parseUrl(req).pathname
+    const parsedPath = sysPath.parse(path)
+
+    // Ignore html files.
```
Comment (KyleAMathews) [inline packages/gatsby/src/commands/develop-static.js]: This is what I added

**Comment by KyleAMathews** [comment]
We have some development server tests? This would be easy to test. Write out a html file to public and then request it and a regular path from the Dev server and verify the two html files are the same.

**Comment by wardpeet** [comment]
How do we feel about the following code?
This way we don't need to keep serve-static in sync, we just do the html check ourselves. rest is handled by static-serve.

```js
const sysPath = require(`path`)
const express = require(`express`)
const parseUrl = require(`parseurl`)

/**
 * Module exports.
 * @public
 */

module.exports = serveStatic

/**
 * @param {string} root
 * @param {object} [options]
 * @return {function}
 * @public
 */

function serveStatic(root, options) {
  const expressStatic = express.static(root, options)

  return function(req, res, next) {
    if ([`GET`, `HEAD`].includes(req.method)) {
      const path = parseUrl(req).pathname
      const parsedPath = sysPath.parse(path)
      if ([`.htm`, `.html`].includes(parsedPath.ext)) {
        return next()
      }
    }

    return expressStatic(req, res, next)
  }
}
```

**Comment by DSchau** [comment]
@kyleamathews @wardpeet that seems like a really good solution! It’s nice to not have to patch and maintain things—so if that works the same we should definitely go that route!

**Comment by KyleAMathews** [comment]
Yeah that'd be much better 👌

**Comment by KyleAMathews** [comment]
Want to just take over the PR?

**Comment by sidharthachatterjee** [comment]
Went ahead and made Ward's changes!

**Comment by DSchau** [comment]
Successfully published:
 - gatsby@2.1.27

**Pull request merged**